### PR TITLE
Use sinon sandboxes by default

### LINKED ISSUE #275: Use sinon sandboxes
We're currently creating mocks/stubs with the global `sinon` module which doesn't do any auto-cleanup. This means we need to ensure all of our mocks/stubs of modules are cleaned up in an `afterEach()`.

We should instead use a [sandbox](http://sinonjs.org/docs/#sandbox) in global `beforeEach()` and `afterEach()` to streamline this.

Once the sandbox is in place we should remove all of the old `.restore()` calls.

See [fireplace](https://github.com/mozilla/fireplace/blob/af44ca5548966abdcb94b3e429022ab4e18a1b00/test-main.js#L5-L18) for an example of how this is done.

### PR DESCRIPTION
**Pull request by mstriemer**
Fixes #275.

### CODE REVIEW

File: test-runner.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require('babel-polyfill');
+require('./tests/client/init');
```
Comment (muffinresearch) [inline test-runner.js]: I think this should work without the `./`

**Comment by muffinresearch** [comment]
r+wc

**Comment by muffinresearch** [comment]
Does this fix: #363?

**Comment by mstriemer** [comment]
This doesn't fix #363 but the error that was happening there is no longer happening on master.

**Comment by muffinresearch** [comment]
Mabye the  `./` removal broke something hence the coverage drop?

**Comment by mstriemer** [comment]
It ran the tests against a merge of this branch and master and `sinon.sanbox.create().sandbox` isn't a thing so there were failing tests.

**Comment by muffinresearch** [comment]
Ah ok, makes sense.

**Pull request merged**